### PR TITLE
add common kubernetes dns names to sans

### LINKED ISSUE #10086: K8s certs assume k8s will be discovered using rancher-dns
we currently generate the certs assuming k8s will be discovered using rancher-dns. We should add sky-dns resolvable fqdn to the list.

Example error message:
```
time=“2017-10-11T17:17:15Z” level=error msg=“Cannot initialize services collection: unable to list Kubernetes services: unable to query any API servers: Get https://kubernetes.default.svc/api/v1/services: x509: certificate is valid for 10.43.0.1, kubernetes, kubernetes.kubernetes, kubernetes.kubernetes.rancher.internal, not kubernetes.default.svc” source=“service.go:117”
```

### PR DESCRIPTION
**Pull request by xetys**
in addition to  #966 to fix https://github.com/rancher/rancher/issues/10086

### CODE REVIEW

**Comment by superseb** [comment]
Thanks, can you rebase this?

**Comment by andrebsguedes** [comment]
The issue apache-spark-on-k8s/spark#543 depends on this!

File: infra-templates/k8s/37/rancher-compose.yml [hunk 1]
```diff
@@ -220,6 +220,10 @@ kubernetes:
         sans:
         - "IP:${KUBEAPI_CLUSTER_IP}"
         - "kubernetes.default.svc.cluster.local"
+        - "kubernetes.default.svc.cluster"
+        - "kubernetes.default.svc"
+        - "kubernetes.default"
```
Comment (alena1108) [inline infra-templates/k8s/37/rancher-compose.yml:225]: these 2 tags - like 225 and 226- are already being added by cattle https://github.com/rancher/cattle/blob/v1.6/code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/service/impl/ServiceDiscoveryApiServiceImpl.java#L855

Comment (alena1108) [reply infra-templates/k8s/37/rancher-compose.yml:225]: so remove `kubernetes.default`and `kubernetes` from the list

**Comment by deniseschannon** [comment]
Closing in favor of https://github.com/rancher/rancher-catalog/pull/1077

@xetys We can only make changes in new catalog entries we are introducing for a version. This will be available in 1.6.15 with k8s 1.9.x

**Pull request closed without merging**